plugins/services/src/js/components/ServiceStatusIcon.tsx

```diff
@@ -34,12 +35,30 @@ const getTooltipContent = (service, content) => {
   );
 };
 
-class ServiceStatusIcon extends Component {
-  getDeclinedOffersWarning(service) {
+type ServiceOrPodOrTree = Service | ServiceTree | Pod;
+
+class ServiceStatusIcon extends React.Component<{
+  showTooltip?: boolean;
+  tooltipContent: JSX.Element;
+  service: ServiceOrPodOrTree;
+}> {
+  static propTypes = {
```

Comment:
do propTypes still make sense when using TS? these are just duplicates for whats in the props interface definition above.

Reply:
as those are runtime-checks that will help us to prevent bugs when using stuff from TS i did not want to remove them. maybe we should gather opinions around that.

Reply:
I'm also in favor of keeping them to help prevent bugs from passing the wrong props from a component that isn't converted to Typescript yet